### PR TITLE
Fixes/rest version update fixes

### PR DESCRIPTION
**Pull request by tekwani**
Updated rest api dependency version and modified all the rest method calls as per the refactoring in the new version.
Uncommented code to return Recurring Algo orders.

### CODE REVIEW

**Review by tarcisiozf — APPROVED** [review]

**Comment by dmytroshch** [comment]
Don't forget to add `recurring` to feature flags https://github.com/bitfinexcom/bfx-hf-server/blob/master/config/feature_flags.json

File: lib/ws_servers/api/handlers/on_settings_update.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ module.exports = async (server, ws, msg) => {
     dmsScope: { type: 'string', v: dmsScope },
     packetWDDelay: { type: 'number', v: packetWDDelay },
     autoResumeAOs: { type: 'bool', v: autoResumeAOs },
-    showOptInModal: { type: 'bool', v: showOptInModal },
+    // showOptInModal: { type: 'bool', v: showOptInModal },
```
Comment (tarcisiozf) [inline lib/ws_servers/api/handlers/on_settings_update.js:51]: commented code can be remove

**Review by tarcisiozf — APPROVED** [review]

File: lib/ws_servers/api/handlers/on_settings_update.js (hunk 1 above)
Comment (tekwani) [reply lib/ws_servers/api/handlers/on_settings_update.js]: fixed.

File: lib/ws_servers/api/handlers/algo_order_params/set.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ module.exports = async (server, ws, msg) => {
     return sendError(ws, 'Unauthorized', ['unauthorized'])
   }
 
-  const err = validateAOParams(algos, algoID, marketData, params)
+  const err = validateAOParams(algos, algoID, marketData, { _symbol: symbol, ...params })
```
Comment (dmytroshch) [inline lib/ws_servers/api/handlers/algo_order_params/set.js:35]:
```suggestion
  const err = validateAOParams(algos, algoID, marketData, { symbol, ...params })
```

File: lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ module.exports = (algos, algoID, marketData, payload) => {
   const symbolDetail = marketData.get(_symbol)
```
Comment (dmytroshch) [inline lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js:21]: let's rename it to `symbol`

File: lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ module.exports = (algos, algoID, marketData, payload) => {
...
 
   const params = _isFunction(processParams)
-    ? processParams(payload, symbolDetail)
+    ? (processParams.length === 2 ? processParams(payload, symbolDetail) : processParams(payload))
```
Comment (tarcisiozf) [inline lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js:24]: why can't we always pass all the arguments and the provided function ignore the additional one?

File: lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js (hunk 3 above)
Comment (tekwani) [reply lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js]: Done.

File: lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js (hunk 4 above)
Comment (tekwani) [reply lib/ws_servers/api/handlers/algo_order_params/validate_ao_params.js]: Updated.

**Review by tarcisiozf — APPROVED** [review]

**Review by tarcisiozf — APPROVED** [review]

**Pull request merged**